### DISCUSSION TITLE
Create a GUI

### DISCUSSION THREAD

**Discussion by hoshsadiq** [knqyf263/pet#282] [Ideas]
It would be nice if this had a GUI option. Specifically I'd like to be able to register a global hot key that brings up a search (+ the recently used snippets) and automatically copies to the clipboard. An option for the GUI would be https://github.com/fyne-io/fyne. Alternatively there are plenty of options for creating GUIs: https://github.com/avelino/awesome-go/blob/master/README.md#gui

**Comment by RamiAwar** [comment]
Okay, let's assume this did exist. Aren't your snippets mainly code snippets? Wouldn't you want to paste them in a terminal?

Or are you using pet for more general things? @hoshsadiq

**Reply by hoshsadiq** [reply]
That was the idea, yes. e.g. I'd like to put random snippets that I often use and automatically paste it in my code editor.

**Reply by RamiAwar** [reply]
I see. Well I'll have to think about it quite a bit. I like the idea, but it does stray away quite a bit from what Pet is, which is a command line snippet manager and not a code snippet manager.

https://github.com/maaslalani/nap is a bit closer to what you're using Pet for, check it out!

But at the same time, I do like the idea of being able to execute Pet from anywhere. I can imagine myself using that in the future, even for command line snippet executions (not just code snippets like you're doing!).

I think I'll look into it a little and get back to you on this.

**Reply by RamiAwar** [reply]
@hoshsadiq What did you end up doing? I'm curious 🧐

**Reply by hoshsadiq** [reply]
Nothing, sorry! I didn't end up using Pet mostly because of this lacking. My current setup is not better though (just put snippets in a folder).

**Reply by RamiAwar** [reply]
Oh okay, let me know if you give nap a try, it seems pretty cool for this usecase!
